src/main/java/com/untamedears/JukeAlert/storage/JukeAlertLogger.java

```diff
@@ -671,21 +663,42 @@ public void saveAllSnitches() {
         return info;
     }
 
-    public List<SnitchAction> getSnitchInfo(int snitchId, int offset) {
+    public List<SnitchAction> getSnitchInfo(int snitchId, int offset, LoggedAction filterType, String filterPlayer) {
         List<SnitchAction> info = new ArrayList<SnitchAction>();
-
         try {
-            getSnitchLogStmt.setInt(1, snitchId);
-            getSnitchLogStmt.setInt(2, offset);
-            getSnitchLogStmt.setInt(3, logsPerPage);
-
-            ResultSet set = getSnitchLogStmt.executeQuery();
-            if (set.isBeforeFirst()) {
+            // statement to get LIMIT entries OFFSET from a number from the snitchesDetailsTbl based on a snitch_id from the main snitchesTbl
+            // optionally it can also be filtered by action type and initiated username
+            // LIMIT ?,? means offset followed by max rows to return
+            String stmt = String.format("SELECT * FROM %s WHERE snitch_id=? AND soft_delete = 0", this.snitchDetailsTbl);
+            if (filterType != null){
+                stmt += " AND snitch_logged_action=?";
+            }
+            if (!filterPlayer.isEmpty()){
+                stmt += " AND snitch_logged_initiated_user LIKE ?";
+            }
+            stmt += " ORDER BY snitch_log_time DESC LIMIT ?,?";
+            PreparedStatement getFilteredSnitchLogStmt = db.prepareStatement(stmt);
+            
+            getFilteredSnitchLogStmt.setInt(1, snitchId);
+            int pos = 2;
+            if (filterType != null){
+                getFilteredSnitchLogStmt.setByte(pos, (byte) filterType.getLoggedActionId());
+                pos++;
+            }
+            if (!filterPlayer.isEmpty()){
+                getFilteredSnitchLogStmt.setString(pos, "%" + filterPlayer + "%");
```

Comment:
I didn't notice anywhere else so, although using a PreparedStatement shelters you from the worst of people's attempts at "injection" attacks, I strongly recommend if nothing else that you limit the valid length of the filterPlayer parameter. It can at most be 16 characters (max length of a player's MC name) so you should truncate any string longer then that, or explicitly fail.